src/LanguageServer/Impl/Implementation/Server.WorkspaceSymbols.cs

```diff
@@ -25,32 +27,40 @@ public sealed partial class Server {
         private static int _symbolHierarchyMaxSymbols = 1000;
 
         public async Task<SymbolInformation[]> WorkspaceSymbols(WorkspaceSymbolParams @params, CancellationToken cancellationToken) {
-            return Array.Empty< SymbolInformation>();
-        }
-
-        public async Task<SymbolInformation[]> DocumentSymbol(DocumentSymbolParams @params, CancellationToken cancellationToken) {
-            return Array.Empty<SymbolInformation>();
+            var symbols = await _indexManager.WorkspaceSymbolsAsync(@params.query,
+                                                                    _symbolHierarchyMaxSymbols,
+                                                                    cancellationToken);
+            return symbols.Select(MakeSymbolInfo).ToArray();
         }
 
         public async Task<DocumentSymbol[]> HierarchicalDocumentSymbol(DocumentSymbolParams @params, CancellationToken cancellationToken) {
-            return Array.Empty<DocumentSymbol>();
+            var path = @params.textDocument.uri.AbsolutePath;
+            var symbols = await _indexManager.HierarchicalDocumentSymbolsAsync(path, cancellationToken);
+            return symbols.MaybeEnumerate().Select(hSym => MakeDocumentSymbol(hSym)).ToArray();
         }
 
+        private static SymbolInformation MakeSymbolInfo(Indexing.FlatSymbol s) {
+            return new SymbolInformation {
+                name = s.Name,
+                kind = (Protocol.SymbolKind)s.Kind,
+                location = new Location {
+                    range = s.Range,
+                    uri = new Uri(s.DocumentPath),
+                },
+                containerName = s.ContainerName,
+            };
+        }
 
-        private static SymbolKind ToSymbolKind(PythonMemberType memberType) {
-            switch (memberType) {
-                case PythonMemberType.Unknown: return SymbolKind.None;
-                case PythonMemberType.Class: return SymbolKind.Class;
-                case PythonMemberType.Instance: return SymbolKind.Variable;
-                case PythonMemberType.Function: return SymbolKind.Function;
-                case PythonMemberType.Method: return SymbolKind.Method;
-                case PythonMemberType.Module: return SymbolKind.Module;
-                case PythonMemberType.Property: return SymbolKind.Property;
-                case PythonMemberType.Union: return SymbolKind.Object;
-                case PythonMemberType.Variable: return SymbolKind.Variable;
-                case PythonMemberType.Generic: return SymbolKind.TypeParameter;
-                default: return SymbolKind.None;
-            }
+        private DocumentSymbol MakeDocumentSymbol(HierarchicalSymbol hSym) {
+            return new DocumentSymbol {
+                name = hSym.Name,
+                detail = hSym.Detail,
+                kind = (Protocol.SymbolKind)hSym.Kind,
```

Comment:
Can we avoid the cast

Reply:
I can make SymbolIndexer use LSP's SymbolKind now that the indexer is on LSP's DLL.
Protocol.SymbolKind and Indexing.SymbolKind where the exact same thing.

Reply:
If this type is going to be removed in general, then every other type the indexer declares could also potentially be removed. They're all nearly copies of the LSP classes, except that before they were structs, so were a bit difficult to work with.

Reply:
(I'm not necessarily advocating for this, just noting that it's possible.)

Reply:
Scratch that comment; That'd mean a refactor back to URIs instead of string paths.

Reply:
I avoided the cast, but adding a giant switch case. I'm not sure if there's a better way

Reply:
That's OK. Although technically numbers may match and all enums are ints, subtle change latee may yield odd bugs when numbers stop being the same. Switch may be large but it is low maintenance code.